### PR TITLE
Implement Hashset

### PR DESCRIPTION
**Pull request by WilliamTakeshi**
Hello!

First of all, thank you for the project. I discovered it through a Reddit post and was really interested in seeing how the implementation came together. I decided to start with the HashSet issue #5  since it looked like the most straightforward to start.

In the HashSet<T> all the methods are the same as HashMap, except for the ones I removed and one that I renamed for clarity.

Please feel free to suggest any changes or give feedback—I’m open to it!

Thanks for your time!

Removed methods:
* compute
* get_key_value
* get_or_insert
* get_or_insert_with
* keys
* remove_entry
* try_insert
* update
* update_or_insert
* update_or_insert_with
* values

Renamed:
* contains_key -> contains

### CODE REVIEW

**Comment by WilliamTakeshi** [comment]
I fixed `cargo fmt` and the check for 1.72

But I will probably need help with the mac stress and the errors here: https://github.com/ibraheemdev/papaya/actions/runs/11555441048/job/32448555759

I don't know what I changed that broke both :(

**Review by ibraheemdev — APPROVED** [review]
This looks amazing, thanks!

Both of those CI errors look unrelated. The macOS stress test failure is a spurious stack overflow that I've been seeing occasionally, I suspect it's just due to the lower default stack size on macOS rather than an infinite recursion error, as the tests are intentionally quite intensive.

The other error is very odd, I'm not sure what's causing it. I'll take a look.

File: src/set.rs [hunk 1]
```diff
@@ -0,0 +1,913 @@
...
+/// [`HashSet::guard`] or using the [`HashSet::pin`] API. See the [crate-level documentation](crate#usage)
+/// for details.
+pub struct HashSet<K, S = RandomState> {
+    raw: raw::HashMap<K, (), S>,
+}
+
+// Safety: We only ever hand out &K/V through shared references to the map,
+// so normal Send/Sync rules apply. We never expose owned or mutable references
+// to keys or values.
+// TODO
```
Comment (ibraheemdev) [inline src/set.rs]: Can this TODO be removed? The bounds look correct to me.

**Comment by ibraheemdev** [comment]
I've fixed the errors on master, this should be good to go now after the TODO is addressed.

File: src/set.rs (hunk 1 above)
Comment (WilliamTakeshi) [reply src/set.rs]: Removed! Sorry for the late reply

**Comment by ibraheemdev** [comment]
Nice, thanks!

**Pull request merged**